File: test-plugins/org.yakindu.sct.simulation.core.sexec.test/test-gen/org/yakindu/sct/simulation/core/sexec/test/SimpleEventTest.java
/**
 * Copyright (c) 2013 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */
package org.yakindu.sct.simulation.core.sexec.test;
import org.eclipse.xtext.junit4.InjectWith;
import org.eclipse.xtext.junit4.XtextRunner;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.yakindu.sct.model.sexec.ExecutionFlow;
import org.yakindu.sct.model.sexec.interpreter.test.util.AbstractExecutionFlowTest;
import org.yakindu.sct.model.sexec.interpreter.test.util.SExecInjectionProvider;
import org.yakindu.sct.test.models.SCTUnitTestModels;
import com.google.inject.Inject;
import static org.junit.Assert.assertTrue;
/**
 *  Unit TestCase for SimpleEvent
 */
@SuppressWarnings("all")
@RunWith(XtextRunner.class)
@InjectWith(SExecInjectionProvider.class)
public class SimpleEventTest extends AbstractExecutionFlowTest {
	@Inject
	private SCTUnitTestModels models;
	@Before
	public void setup() throws Exception {
		ExecutionFlow flow = models
				.loadExecutionFlowFromResource("SimpleEvent.sct");
		initInterpreter(flow);
	}
	@Test
	public void simpleEventTest() throws Exception {
		interpreter.enter();
		assertTrue("Expected A to be active", isActive("A"));
		assertTrue(5 == 5);
		raiseEvent("Event1");
		interpreter.runCycle();
		assertTrue("Expected B to be active", isActive("B"));
	}
}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/SimpleEventTest/SimpleEventTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "SimpleEvent.h"

SimpleEvent handle;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		simpleEvent_init(&handle);
	}
};


TEST_F(StatemachineTest, simpleEventTest) {					
	simpleEvent_enter(&handle);
	EXPECT_TRUE(simpleEvent_isStateActive(&handle, SimpleEvent_main_region_A)) << "Expected A to be active" ;
	EXPECT_TRUE(5l== 5l);
	simpleEventIface_raise_event1(&handle);
	simpleEvent_runCycle(&handle);
	EXPECT_TRUE(simpleEvent_isStateActive(&handle, SimpleEvent_main_region_B)) << "Expected B to be active" ;
	simpleEvent_runCycle(&handle);
	EXPECT_TRUE(!simpleEvent_isStateActive(&handle, SimpleEvent_main_region_B));
}


